analyzer/src/environment/variables.rs

```diff
@@ -130,7 +147,9 @@ impl Locals {
     /// # Panics
     /// This method panics if the maximum number of scopes has been reached.
     fn begin_scope(&mut self) {
-        self.current_depth = self.current_depth.checked_add(1).expect("Too many scopes");
+        self.current_depth = (self.current_depth as isize)
+            .checked_add(1)
+            .expect("Too many scopes") as usize;
```

Comment:
The maximum number of scopes has been divided by two, it's now `usize::MAX / 2` for the sign bit, instead of `usize::MAX - 1`. Doing a checked add on the `isize` type, the type used in `Variable::depth`, avoids further repercussions further down. `usize::MAX` is not representable as an `isize` for instance.

Reply:
👍🏻 my bad